src/editor/components/preview/data.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * ---------------------------------------------------------------------
+ * Copyright (c) 2020 EclipseSource Munich
+ * Licensed under MIT
+ * https://github.com/eclipsesource/jsonforms-editor/blob/master/LICENSE
+ * ---------------------------------------------------------------------
+ */
+import { JsonSchema } from '@jsonforms/core';
+
+export const generateDataForPreview = (
+  schema: JsonSchema,
+  data: any = {}
+): object => {
+  if (schema && schema.type === 'object' && schema.properties) {
+    Object.entries(schema.properties).forEach(([key, value]) => {
+      if (value?.type === 'object') {
+        data[key] = generateDataForPreview(value, {});
+      }
+    });
+  }
+  return data;
+};
```

Comment:
I would place the code in `schema.ts` and let it work on our intermediate format, not the "real/export" format.

- All schema related logic is then in the same place
- All edge cases are then implicitly handled, as for example an object doesn't need to have `type: 'object'`, it could also only have `properties` which automatically makes it to an object.

Reply:
Moved the method to `schema.ts`